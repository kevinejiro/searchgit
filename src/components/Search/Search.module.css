.container {
  background-color: transparent;
  border-radius: 2.5rem;
  height: 4.1rem;
  display: flex;
  align-items: center;
  padding: 0 0rem 0 2.2rem;
  margin: 0 2.4rem 2.1rem;
  border: 1px solid var(--border-color);
  width: 80%;
  transition: all 0.3s ease-in-out;
  justify-content: center;
  z-index: 5;
  position: absolute;
}

.container:hover,
.container.active {
  background: var(--input-bg-hover-color) 0% 0% no-repeat padding-box;
}

.container img {
  height: 2rem;
  width: 2rem;
}

.container input {
  background-color: transparent;
  border: none;
  outline: none;
  font-family: Metropolis;
  font-weight: 400;
  font-size: 1.3rem;
  width: 18.6rem;
  height: inherit;
  color: var(--input-text-color);
  flex: 1;
}
.container input::placeholder {
  color: var(--input-text-color);
  font-family: Metropolis;
  font-weight: 400;
  font-size: 1.4rem;
  opacity: 0.6;
}
.container button {
  background: transparent
    linear-gradient(
      var(--button-background-degree),
      var(--button-background-gray1) 0%,
      var(--button-background-gray2) 100%
    )
    0% 0% no-repeat padding-box;
  color: var(--button-text);
  font-family: Metropolis;
  font-weight: lighter;
  font-size: 1.4rem;
  border: none;
  border-radius: 2.5rem;
  height: 4.1rem;
  width: 8.4rem;
  mix-blend-mode: normal;
  transition: all 0.3s ease-in-out;
}
.container button:hover {
  cursor: not-allowed;
}

.container button.active {
  background: var(--button-background-active) 0% 0% no-repeat padding-box;
  color: var(--button-text-active);
  cursor: pointer;
}

.count {
  margin: 0 1rem;
  color: var(--button-background-active);
  font-size: 1rem;
  font-weight: 700;
  line-height: normal;
  text-align: center;
}

.error-message {
  margin: 0 1rem;
  color: #c3002d;
  font-size: 1rem;
  line-height: normal;
  text-align: center;
}
.user-not-found-input {
  width: 100% !important;
}

.search-icon-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1%;
  position: absolute;
  top: -8rem;
}

.search-icon {
  width: 20%;
}

.search-icon-text {
  font: normal normal 300 22px/30px Metropolis;
  letter-spacing: 0px;
  color: var(--input-text-color);
  margin-left: 0.4rem;
  text-decoration: none;
}

.search-icon-text > span {
  font: normal normal bold 26px/31px Metropolis;
}

/* Tablet */
@media (min-width: 768px) {
  .container {
    width: 35%;
    height: 5rem;
    padding: 0 0rem 0 2.2rem;
    margin: 0 7.85rem 2.1rem;
    max-width: 49.688rem;
  }
  .container input {
    font-size: 1.8rem;
    height: inherit;
  }
  .container button {
    height: 5rem;
    width: 10.6rem;
    border-radius: 2.5rem;
  }
  .error-message {
    font-size: 1.5rem;
  }
  .search-icon {
    width: 25%;
  }
}

/* Desktop */
@media (min-width: 1024px) {
  .search-icon {
    width: 25%;
  }
  .container {
    width: 35%;
    margin: 0 0 2.4rem;
  }
  .container button {
    font-size: 1.6rem;
  }
}
