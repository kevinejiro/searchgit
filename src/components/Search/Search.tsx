import classes from "./Search.module.css";
import GitIcon from "../common/GitIcon/index";
import { motion } from "framer-motion";
import React, { useState } from "react";
import { useHistory } from "react-router-dom";

import { ctx } from "../../App";

export type Props = {
  ctx: ctx;
};

const Search: React.FC<Props> = ({ ctx }) => {
  let history = useHistory();
  const [enteredUsername, setEnteredUsername] = useState(ctx?.username);

  const searchHandler = (e: React.FormEvent) => {
    e.preventDefault();
    if (enteredUsername.trim() !== "") {
      ctx.setUsername(enteredUsername);
      history.push("/results");
    }
  };

  return (
    <form
      className={`${classes["container"]} ${
        classes[`${enteredUsername.trim() !== "" && "active"}`]
      }`}
      onSubmit={searchHandler}
      autoComplete="off"
    >
      <motion.div
        initial={{ y: "5rem", opacity: 0 }}
        animate={{ y: "0rem", opacity: 1 }}
        exit={{ opacity: 0 }}
        className={classes["search-icon-wrapper"]}
      >
        <div className={classes["search-icon"]}>
          <GitIcon />
        </div>
        <div className={classes["search-icon-text"]}>
          search<span>Git</span>
        </div>
      </motion.div>
      <input
        className={ctx?.users.length > 0 ? classes["user-not-found-input"] : ""}
        type="text"
        name="username"
        autoFocus
        value={enteredUsername}
        placeholder="Search GitHub usernameâ€¦"
        onChange={(e) => setEnteredUsername(e.target.value)}
      />
      <button
        className={enteredUsername.trim() !== "" ? classes["active"] : ""}
      >
        Submit
      </button>
    </form>
  );
};

export default Search;
