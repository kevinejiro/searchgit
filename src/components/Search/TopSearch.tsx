import { motion } from "framer-motion";
import React, { useState } from "react";

// type
import { ctx } from "../../App";

// css import
import classes from "./Search.module.css";

type Props = {
  ctx: ctx;
};

const transition = { duration: 0.6, ease: ["easeOut"] };

const TopSearch: React.FC<Props> = ({ ctx }) => {
  const [enteredUsername, setEnteredUsername] = useState(ctx?.username);

  // this updates setUsername state in context
  const searchHandler = (e: React.FormEvent) => {
    e.preventDefault();
    if (enteredUsername.trim() !== "") {
      ctx.setUsername(enteredUsername);
    }
  };

  return (
    <motion.form
      className={`${classes["container"]} ${
        classes[`${enteredUsername.trim() !== "" && "active"}`]
      }`}
      onSubmit={searchHandler}
      autoComplete="off"
      transition={transition}
      initial={{ top: "50%", zIndex: 1, position: "absolute" }}
      animate={{ top: "4%", zIndex: 4, position: "absolute" }}
      exit={{ top: "50%", zIndex: 1, position: "absolute" }}
    >
      <input
        className={
          ctx?.errorMessage !== "" && ctx?.users.length > 0
            ? classes["user-not-found-input"]
            : ""
        }
        type="text"
        name="username"
        autoFocus
        value={enteredUsername}
        placeholder="Search GitHub usernameâ€¦"
        onChange={(e) => setEnteredUsername(e.target.value)}
      />
      {ctx?.errorMessage === "" &&
        ctx?.users.length > 0 &&
        enteredUsername !== "" && (
          <p className={classes["count"]}>
            {ctx?.usersTotalCount}{" "}
            {ctx?.usersTotalCount > 1 ? "results" : "result"}
          </p>
        )}
      <button
        className={enteredUsername.trim() !== "" ? classes["active"] : ""}
      >
        Submit
      </button>
    </motion.form>
  );
};

export default TopSearch;
