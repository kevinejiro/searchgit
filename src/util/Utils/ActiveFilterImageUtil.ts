import contributorsImageActive from "../../assets/images/contributorsActive.svg";
import repoActive from "../../assets/images/repoActive.svg";
import contributorsImage from "../../assets/images/contributors.svg";
import repoImage from "../../assets/images/repo.svg";

const ActiveFilterImageUtil = {
  getImageList(input: string) {
    let contributionsImageUrl;
    let repositoriesImageUrl;
    let urlList;
    switch (input) {
      case "user":
        contributionsImageUrl = contributorsImageActive;
        repositoriesImageUrl = repoImage;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
      case "org":
        contributionsImageUrl = contributorsImage;
        repositoriesImageUrl = repoActive;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
      default:
        contributionsImageUrl = contributorsImage;
        repositoriesImageUrl = repoImage;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
    }
  },
};

export default ActiveFilterImageUtil;
