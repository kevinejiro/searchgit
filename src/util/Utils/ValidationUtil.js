const ValidationUtil = {
	nullChecker(input) {
		if (input !== undefined && input !== null) {
			return input;
		}

		return "N/A";
	},
};

export default ValidationUtil;
