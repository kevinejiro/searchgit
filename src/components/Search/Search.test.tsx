import { render } from "../../util/Utils/test-utils";
// import userEvent from "@testing-library/user-event";

import Search, { Props } from "./Search";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe("<Pagination />", () => {
  it("renders the component without error", () => {
    const mockProps: Props = {
      ctx: {
        users: [],
        errorMessage: "",
        usersTotalCount: 10,
        username: "kevin",
        isEmptyList: false,
        isLoading: false,
        filterParam: "",
        page: 1,
        setUsername: jest.fn(),
        setFilterParam: jest.fn(),
        setPage: jest.fn(),
        pageCount: 1,
      },
    };
    const component = render(<Search {...mockProps} />);
    expect(component).toMatchSnapshot();

    // expect(mockProps.handlePagination).toBeCalledWith(2);
    // wrapper.getByRole("button").at(2).simulate("click");
    // expect(mockProps.handlePagination).toBeCalledWith(3);w
    // wrapper.getByRole("button").at(3).simulate("click");
    // expect(mockProps.handlePagination).toBeCalledWith(5);
    // wrapper.getByRole("button").at(4).simulate("click");
    // expect(mockProps.handlePagination).toBeCalledWith(2);
    // expect(wrapper.render()).toMatchSnapshot();
  });
  it("renders the component from the end", () => {
    const mockProps: Props = {
      ctx: {
        users: [],
        errorMessage: "",
        usersTotalCount: 10,
        username: "kevin",
        isEmptyList: false,
        isLoading: false,
        filterParam: "",
        page: 1,
        setUsername: jest.fn(),
        setFilterParam: jest.fn(),
        setPage: jest.fn(),
        pageCount: 1,
      },
    };

    const component = render(<Search {...mockProps} />);
    expect(component).toBeDefined();
  });
});
