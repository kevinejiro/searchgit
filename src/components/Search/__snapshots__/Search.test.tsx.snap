// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders the component without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
        class="container active"
      >
        <div
          class="search-icon-wrapper"
          style="opacity: 0; transform: translateY(5rem) translateZ(0);"
        >
          <div
            class="search-icon"
          >
            <svg
              class="svg"
              height="69"
              viewBox="0 0 66 69"
              width="66"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Group 30"
                id="Group_30"
                transform="translate(-279.361 -224.819)"
              >
                <path
                  d="M61.828,68.277,47.162,53.491a2.678,2.678,0,0,1-.227-.262,29.055,29.055,0,0,1-17.6,5.914A29.573,29.573,0,1,1,58.668,29.57a29.6,29.6,0,0,1-8.092,20.395l.042.042L65.285,64.792a2.48,2.48,0,0,1,0,3.484,2.427,2.427,0,0,1-3.457,0ZM4.889,29.57A24.446,24.446,0,1,0,29.333,4.928,24.572,24.572,0,0,0,4.889,29.57ZM24.7,43.6a14.673,14.673,0,1,1,9.277-.005c-.741.146-.975-.314-.975-.71V38.828a3.53,3.53,0,0,0-1.009-2.736c3.266-.366,6.7-1.615,6.7-7.294a5.736,5.736,0,0,0-1.509-3.969,5.359,5.359,0,0,0-.146-3.912s-1.226-.4-4.029,1.515a13.851,13.851,0,0,0-7.346,0c-2.8-1.91-4.03-1.515-4.03-1.515a5.37,5.37,0,0,0-.144,3.918,5.736,5.736,0,0,0-1.51,3.969c0,5.663,3.421,6.931,6.678,7.3a3.172,3.172,0,0,0-.93,1.978,3.11,3.11,0,0,1-4.268-1.23,3.077,3.077,0,0,0-2.249-1.525s-1.43-.019-.1.9a3.908,3.908,0,0,1,1.63,2.162s.849,2.639,4.929,1.742v2.753a.678.678,0,0,1-.716.738A1.373,1.373,0,0,1,24.7,43.6Z"
                  data-name="Union 1"
                  fill="#27d3a5"
                  id="Union_1"
                  transform="translate(279.361 224.819)"
                />
              </g>
            </svg>
          </div>
          <div
            class="search-icon-text"
          >
            search
            <span>
              Git
            </span>
          </div>
        </div>
        <input
          class=""
          name="username"
          placeholder="Search GitHub username…"
          type="text"
          value="kevin"
        />
        <button
          class="active"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
      class="container active"
    >
      <div
        class="search-icon-wrapper"
        style="opacity: 0; transform: translateY(5rem) translateZ(0);"
      >
        <div
          class="search-icon"
        >
          <svg
            class="svg"
            height="69"
            viewBox="0 0 66 69"
            width="66"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Group 30"
              id="Group_30"
              transform="translate(-279.361 -224.819)"
            >
              <path
                d="M61.828,68.277,47.162,53.491a2.678,2.678,0,0,1-.227-.262,29.055,29.055,0,0,1-17.6,5.914A29.573,29.573,0,1,1,58.668,29.57a29.6,29.6,0,0,1-8.092,20.395l.042.042L65.285,64.792a2.48,2.48,0,0,1,0,3.484,2.427,2.427,0,0,1-3.457,0ZM4.889,29.57A24.446,24.446,0,1,0,29.333,4.928,24.572,24.572,0,0,0,4.889,29.57ZM24.7,43.6a14.673,14.673,0,1,1,9.277-.005c-.741.146-.975-.314-.975-.71V38.828a3.53,3.53,0,0,0-1.009-2.736c3.266-.366,6.7-1.615,6.7-7.294a5.736,5.736,0,0,0-1.509-3.969,5.359,5.359,0,0,0-.146-3.912s-1.226-.4-4.029,1.515a13.851,13.851,0,0,0-7.346,0c-2.8-1.91-4.03-1.515-4.03-1.515a5.37,5.37,0,0,0-.144,3.918,5.736,5.736,0,0,0-1.51,3.969c0,5.663,3.421,6.931,6.678,7.3a3.172,3.172,0,0,0-.93,1.978,3.11,3.11,0,0,1-4.268-1.23,3.077,3.077,0,0,0-2.249-1.525s-1.43-.019-.1.9a3.908,3.908,0,0,1,1.63,2.162s.849,2.639,4.929,1.742v2.753a.678.678,0,0,1-.716.738A1.373,1.373,0,0,1,24.7,43.6Z"
                data-name="Union 1"
                fill="#27d3a5"
                id="Union_1"
                transform="translate(279.361 224.819)"
              />
            </g>
          </svg>
        </div>
        <div
          class="search-icon-text"
        >
          search
          <span>
            Git
          </span>
        </div>
      </div>
      <input
        class=""
        name="username"
        placeholder="Search GitHub username…"
        type="text"
        value="kevin"
      />
      <button
        class="active"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
